### ISSUE TITLE
2023.3 - Camera Widget transparent (no image)

### ISSUE THREAD

**Issue by Toukite** [home-assistant/android#3443]
**Home Assistant Android app version(s):**2023.3.0-full

**Android version(s):**13

**Device model(s):**OnePlus 8 Pro

**Home Assistant version:**2023.3.6

**Last working Home Assistant release (if known):**2023.3.6 (issue appeared with today's android update, not HA update)

**Description of problem, include YAML if issue is related to notifications:**
Following today's update, camera widgets are not working anymore !
Widget is there but fully transparent, no image.
I tried force cleaning cache, recreating widgets, reboot smartphone without success

**Companion App Logs:**

```

```

**Screenshot or video of problem:**

**Additional information:**
Probably related to https://github.com/home-assistant/android/pull/3375

**Comment by jpelgrom** [comment]
If you tap on the widget, do you get any indication it is refreshing or refreshing fails?

Logs would be helpful as will hopefully indicate why it's failing. Open the app first, then add a widget, and grab the logs from Settings > Companion app > Show and share logs.

**Comment by dshokouhi** [comment]
I am personally unable to reproduce, was able to add another camera widget and it loads as expected

**Comment by CodexHere** [comment]
Also having this problem. I removed the widget, but `Manage Widgets` still shows it (and why can't I remove it from there????)

On boot it shows the HA logo, after widgets fully initialize it's fully transparent. Resizing/etc has no effect on visual output, and there's no errors/warnings in the logs

Companion App Logs: https://pastebin.com/raw/KZFEy0SG

**Comment by dshokouhi** [comment]
> I removed the widget, but Manage Widgets still shows it (and why can't I remove it from there????)

you have to first edit in order to delete it

These logs do not contain anything from our widget code base. At a minimum I would expect to see the below log line. Have you tried tapping on the widget during this test like mentioned [above](https://github.com/home-assistant/android/issues/3443#issuecomment-1491884021)?

https://github.com/home-assistant/android/blob/master/app/src/main/java/io/homeassistant/companion/android/widgets/camera/CameraWidget.kt#L149

**Comment by CodexHere** [comment]
Yep I tapped it. And let me more clear - I removed it from my desktop as a widget, but it still shows in the `Manage Widgets` section of the app.

**Comment by dshokouhi** [comment]
> Yep I tapped it.

Can you please open the app, tap on the widget and then grab the companion app logs so we can see what happens then? We should be seeing some of our log data there but your logs above do not show anything there.

> And let me more clear - I removed it from my desktop as a widget, but it still shows in the `Manage Widgets` section of the app.

That is actually a separate issue from this issue actually as this issue is related to loading of the image only. If you can reliably reproduce that issue please open a new issue for it.

**Comment by CodexHere** [comment]
> Can you please open the app, tap on the widget and then grab the companion app logs so we can see what happens then? We should be seeing some of our log data there but your logs above do not show anything there.

That's exactly what you're seeing. Those are the logs form exactly that behavior.

**Comment by dshokouhi** [comment]
unfortunately those logs dont even show that you tapped on the widget

**Comment by CodexHere** [comment]
Also, there are 2 instances of the widget in the `Manage Widgets` section when I have readded the widget. So there's something else going on as well.

> unfortunately those logs dont even show that you tapped on the widget

To me, that would imply a deeper issue with the widget then. I can't tap any harder or softer, to make it magically log.

**Comment by dshokouhi** [comment]
> Also, there are 2 instances of the widget in the `Manage Widgets` section when I have readded the widget. So there's something else going on as well.

lets please not describe multiple issues in this one as it can become confusing its best to focus on the widget that is not updating.



> > unfortunately those logs dont even show that you tapped on the widget
> 
> To me, that would imply a deeper issue with the widget then. I can't tap any harder or softer, to make it magically log.

actually Android notifies the app to perform an update and we are not seeing that here. Maybe a start fresh is needed in your case?

https://companion.home-assistant.io/docs/troubleshooting/faqs/#starting-fresh-with-the-android-app

**Comment by CodexHere** [comment]
I literally can't provide any more proof or evidence than this... 

Here's a screen recording: https://streamable.com/y5seuk

And here's the logs after all that: https://pastebin.com/raw/nkwA1PgU

**Comment by Toukite** [comment]
Hello,

On my side I see following stacktrace in companion logs when I add the widget:

`03-31 19:19:40.771 20238 22399 E SensorReceiver: Error while getting core config to sync sensor status
03-31 19:19:40.771 20238 22399 E SensorReceiver: io.homeassistant.companion.android.common.data.integration.IntegrationException: com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
03-31 19:19:40.771 20238 22399 E SensorReceiver:  at [Source: (okhttp3.ResponseBody$BomAwareReader); line: 1, column: 0]
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at io.homeassistant.companion.android.common.data.integration.impl.IntegrationRepositoryImpl.getConfig(IntegrationRepositoryImpl.kt:493)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at io.homeassistant.companion.android.common.data.integration.impl.IntegrationRepositoryImpl$getConfig$1.invokeSuspend(Unknown Source:14)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
03-31 19:19:40.771 20238 22399 E SensorReceiver: Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
03-31 19:19:40.771 20238 22399 E SensorReceiver:  at [Source: (okhttp3.ResponseBody$BomAwareReader); line: 1, column: 0]
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1741)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at com.fasterxml.jackson.databind.ObjectReader._initForReading(ObjectReader.java:359)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:2041)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1459)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at retrofit2.converter.jackson.JacksonResponseBodyConverter.convert(JacksonResponseBodyConverter.java:33)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at retrofit2.converter.jackson.JacksonResponseBodyConverter.convert(JacksonResponseBodyConverter.java:23)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at retrofit2.OkHttpCall.parseResponse(OkHttpCall.java:243)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:153)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
03-31 19:19:40.771 20238 22399 E SensorReceiver: 	at java.lang.Thread.run(Thread.java:1012)
03-31 19:19:40.832 20238 22399 D ServerConnectionInfo: localUrl is: false, usesInternalSsid is: false, usesWifi is: true
03-31 19:19:40.915 20238 22399 W SensorReceiver: Exception while updating sensors: IntegrationException: com.fasterxml.jackson.databind.exc.MismatchedInputException
03-31 19:19:40.916 20238 22399 I SensorReceiver: Sensor updates and sync completed
`

Hope it will help !

**Comment by jpelgrom** [comment]
@Toukite That is a handled error related to sensor updates, so not related. It does show that the response unexpectedly ended, suggesting connection errors. Can you access your dashboard from the app correctly? Are there any other errors in the logs?

**Comment by dshokouhi** [comment]
For those having issues here can you please let us know about the Camera being used? Also by chance what size the images are from the snapshot of the camera? Trying to make sure we capture all data points here.

Also if another user can try to get us the logs that would be great too.

**Comment by Toukite** [comment]
Hello,

There is another error that is repeating a looot with different sensor names (wifi ssid, proximity sensor, steps_sensor, etc etc etc)

`SensorReceiver: Issue registering sensor total_tx_gb
04-02 23:52:29.642 18189 18227 E SensorReceiver: io.homeassistant.companion.android.common.data.integration.IntegrationException: Error calling integration request register_sensor
04-02 23:52:29.642 18189 18227 E SensorReceiver: 	at io.homeassistant.companion.android.common.data.integration.impl.IntegrationRepositoryImpl.registerSensor(IntegrationRepositoryImpl.kt:666)
04-02 23:52:29.642 18189 18227 E SensorReceiver: 	at io.homeassistant.companion.android.common.data.integration.impl.IntegrationRepositoryImpl$registerSensor$1.invokeSuspend(Unknown Source:15)
04-02 23:52:29.642 18189 18227 E SensorReceiver: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
04-02 23:52:29.642 18189 18227 E SensorReceiver: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
04-02 23:52:29.642 18189 18227 E SensorReceiver: 	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
04-02 23:52:29.642 18189 18227 E SensorReceiver: 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
04-02 23:52:29.642 18189 18227 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
04-02 23:52:29.642 18189 18227 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
04-02 23:52:29.642 18189 18227 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
04-02 23:52:29.642 18189 18227 E SensorReceiver: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
04-02 23:52:29.642 18189 18227 D SensorReceiver: Nothing to update for server 1 ()`

When I try to refresh the widget multiple times by taping it (although I can't see it unless I long press to move it), I sometimes have this crash message:
![Screenshot_2023-04-02-23-51-57-01_b783bf344239542886fee7b48fa4b892](https://user-images.githubusercontent.com/42780622/229381159-d37e83ed-0196-4e59-ad7b-11a378e73b00.jpg)


Something I noticed though, it seems that all my widgets are down since the update, not just the camera ones.
Indeed, I have service buttons that I used to call scripts, and now when I click on them the action is not triggered, widget goes red and I have this message:
![Screenshot_2023-04-02-23-52-21-59_b783bf344239542886fee7b48fa4b892](https://user-images.githubusercontent.com/42780622/229381206-ee912859-7401-4417-8eb9-59699fe6ad01.jpg)

Is it possible for me to downgrade the Android app until this can be fixed ?

**Comment by dshokouhi** [comment]
> There is another error that is repeating a looot with different sensor names (wifi ssid, proximity sensor, steps_sensor, etc etc etc)

This is error is unrelated to this issue. Lets not confuse this issue amongst other problems.



> Something I noticed though, it seems that all my widgets are down since the update, not just the camera ones.
> Indeed, I have service buttons that I used to call scripts, and now when I click on them the action is not triggered, widget goes red and I have this message:

we need to see the companion app error from those failures. The failure will even have `widget` mentioned. Its not easy for us to look at small log snippets of random errors. Please reproduce the error and grab the whole set of logs so we can see what is failing.



> Is it possible for me to downgrade the Android app until this can be fixed ?

nothing stopping you from uninstalling the app and grabbing an APK off the github releases tab but in terms of troubleshooting its not going to move this bug further. Developers are unable to reproduce the issue and need logs of the actual issue in order to fix it.

I would like to see logs where the widget makes an attempt to load an image and I am not seeing that yet.

> I don't how to find snapshot details

your camera tech specs will contain this data

**Comment by Toukite** [comment]
Hello,

I'll try to provide full logs but I saw personal data in them so I have to filter, it'll take time
Unfortunately, it's the same as @CodexHere, I did not find any line containing widget keyword, the issue seems to be higher than that.
I don't know how to help more... and only few people seem to reproduce so I suppose I'll have to live with that...
Unless perhaps someone could analyze the code changes of this release more globally (not just the camera PR) to see if any change could be the root cause ? Perhaps an authorization issue ?

This morning I found another thing that's broken: I have notification with action buttons for the alarm, and now the action button are not working either.

In fact, it seems that everything "outside" the app (widgets, notif actions, ...) is broken and can't communicate with HA.
I'll change the issue title since it's more global than just camera widget issue.

**Comment by dshokouhi** [comment]
Guys please do not combine multiple issues into one. We will close this issue if it gets out of hand. We need to stick to 1 issue per report. Please change back the edits.

**Comment by Toukite** [comment]
It's all the same issue, just multiple effects discovered over time... 

Fyi I did a full phone reset today without success: delete mobile entities in ha, restart ha, delete android data and app, reboot phone, clean obsolete data, reinstall app, reconfigure app, ...

**Comment by dshokouhi** [comment]
> It's all the same issue, just multiple effects discovered over time...

I understand you may be experiencing multiple issues but we need to treat each one separately unless we see the same exact error in the logs which we have not seen.  We need to stop combining multiple issues into one unless the developers mention it is indeed related. As of now the 2 errors you shown us are completely unrelated to this issue.



> Fyi I did a full phone reset today without success: delete mobile entities in ha, restart ha, delete android data and app, reboot phone, clean obsolete data, reinstall app, reconfigure app, ...

I don't know where you got those steps but they are incorrect.  You should be following the start fresh steps from troubleshooting. As you have done a reinstall which are not part of the start fresh steps you are already restoring bad data that the steps are designed to avoid. Please make sure to follow each step precisely without skipping or adding a step.

https://companion.home-assistant.io/docs/troubleshooting/faqs/#starting-fresh-with-the-android-app

Please get us the full logs for the camera widget issue. Make sure you reproduce the issue and please convey if the app is crashing or not. We are not getting enough information in this issue and its starting to get out of hand with the separate issues being mentioned. We want to help but it is not easy when multiple issues are getting combined into one.

If you see any other issue please file them separately and do not mention them here to avoid confusing the actual issue with camera images not loading.

**Comment by Toukite** [comment]
I did follow the procedure, it is not working.
If you want us to provide relevant logs, please ensure the app is indeed writing those relevant logs.... 2 people told you there is nothing and provided every possible input.
Please read the symptoms again, you might see the link between them after all ...
From my window you should be reading the code changes of this release and try to reproduce yourself, but no, let's hide behind "github issue bureaucracy" and close the issue, it's probably easier if not better !
I'll let others fight, hoping this issue gets visibility over time.
Thanks for the time anyway.

**Comment by dshokouhi** [comment]
> I did follow the procedure, it is not working.

"it is not working" is not going to help us to troubleshoot the issue, where are the logs?



> If you want us to provide relevant logs, please ensure the app is indeed writing those relevant logs....

We are printing relevant logs, what is not helpful is not knowing if the application is crashing before the logs are retrieved. We depend on the user for that feedback. If the app does not stay open while you create a widget and tap to update it then by not relaying that information to the developer we are not able to help. If the app is indeed crashing then we will need to use a different method to get the logs but without knowing exactly what is happening we are not able to proceed.



> 2 people told you there is nothing and provided every possible input.

I have only seen logs from 1 user. The logs you have sent us (twice now) are not the same logs we are looking for.



> Please read the symptoms again, you might see the link between them after all ...

We have read them multiple times and are still unable to reproduce the issue which is exactly why we are asking for logs from the problem devices.



> From my window you should be reading the code changes of this release and try to reproduce yourself

We have looked at the code changes and there was only 1 change associated to camera widgets. Everything else you mention is unrelated and should be treated like a separate issue with the requested logs attached to the issue. Developers cannot fix an issue that has no logs and is not reproducible.



> but no, let's hide behind "github issue bureaucracy" and close the issue, it's probably easier if not better !

We have an issue to do triage and we are requesting more information, if you are unwilling to provide us with the necessary details then we will not be able to help. This issue was closed because too many other issues were being added to this one after we had requested not to do so. Issues are not meant to be a catch all, they are to report a precise problem and users need to work with developers to get us the requested information. Otherwise we simply cannot proceed.

**Comment by Toukite** [comment]
`what is not helpful is not knowing if the application is crashing before the logs are retrieved.`
Yes, the application is crashing before the logs are retrieved (as you could see from @CodexHere video and my screenshot and the lack of logs, but nevermind...)

`If the app is indeed crashing then we will need to use a different method to get the logs`
I'm all ears.

**Comment by dshokouhi** [comment]
> `If the app is indeed crashing then we will need to use a different method to get the logs`
> I'm all ears.

You will need to grab the logs using either Android Studio or an app like Logcat Reader. Both of which will require special USB permissions to hook up to a computer to get the logs. The easiest of the 2 is Android Studio as you can view the logcat window to copy the crash.

For this after hooking up to the debugger reproduce the issue (only tap 1 time instead of multiple) and look in the logs. We should at a minimum see a log line indicating that the image is being attempted to be downloaded.

https://github.com/home-assistant/android/blob/master/app/src/main/java/io/homeassistant/companion/android/widgets/camera/CameraWidget.kt#L149

The crash should be somewhere around there, probably just after the above log line.

**Comment by Toukite** [comment]
Hello,

OK, so good news today it seems that both following issues are solved:
- service button widget working again
- notification action working again
Perhaps due to the app fresh start of yesterday..

Regarding the last issue, I installed Android Studio and was able to retrieve following stacktrace which I think will be helpful:
```
FATAL EXCEPTION: main
Process: io.homeassistant.companion.android, PID: 21572
java.lang.IllegalArgumentException: **RemoteViews for widget update exceeds maximum bitmap memory usage** (used: 29491200, max: 15396480)
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3044)
	at android.os.Parcel.createException(Parcel.java:3024)
	at android.os.Parcel.readException(Parcel.java:3007)
	at android.os.Parcel.readException(Parcel.java:2949)
	at com.android.internal.appwidget.IAppWidgetService$Stub$Proxy.updateAppWidgetIds(IAppWidgetService.java:820)
	**at android.appwidget.AppWidgetManager.updateAppWidget(AppWidgetManager.java:540)**
	at com.squareup.picasso.RemoteViewsAction$AppWidgetAction.update(RemoteViewsAction.java:116)
	at com.squareup.picasso.RemoteViewsAction.complete(RemoteViewsAction.java:45)
	at com.squareup.picasso.Picasso.deliverAction(Picasso.java:576)
	at com.squareup.picasso.Picasso.complete(Picasso.java:528)
	at com.squareup.picasso.Picasso$1.handleMessage(Picasso.java:122)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:240)
	at android.os.Looper.loop(Looper.java:351)
	at android.app.ActivityThread.main(ActivityThread.java:8364)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.appwidget.AppWidgetServiceImpl.updateAppWidgetInstanceLocked(AppWidgetServiceImpl.java:1917)
	at com.android.server.appwidget.AppWidgetServiceImpl.updateAppWidgetIds(AppWidgetServiceImpl.java:1732)
	at com.android.server.appwidget.AppWidgetServiceImpl.updateAppWidgetIds(AppWidgetServiceImpl.java:1467)
	at com.android.internal.appwidget.IAppWidgetService$Stub.onTransact(IAppWidgetService.java:399)
	at android.os.Binder.execTransactInternal(Binder.java:1326)
```

**Comment by Extreme79** [comment]
I have the exact same issue, with a similar phone, oneplus 8t (no mods, full stock, no root).
After the latest update my camera widgets turn transparent, invisible (only perimeter is visible during editing).
Then android os report that home assistant is crashing repeatedly.

**Comment by dshokouhi** [comment]
Submitted a fix however since I cannot reproduce the issue I would appreciate it if someone impacted by the issue can test the fix. 

To test the fix please download the artifact from: https://github.com/home-assistant/android/actions/runs/4608412941

Extract the artifact and install the debug APK for the phone on the device having issues. The debug APK can be installed side by side with the production app and is safe to remove after the test. Recreate the camera widget using the red debug icon and check if the image loads. You will see a pop-up complaining about notifications after login but that is safe to ignore as Firebase is not available in the debug builds.

**Comment by Extreme79** [comment]
Installed on my oneplus 8t.
It works! Camera widget now show the stream and update itself when clicked.
Thanks!

**Comment by dshokouhi** [comment]
Thanks for checking @Extreme79  so the issue is that some cameras have a very large image, larger than what the device will accept and thus crash before our code can be reached. The old behavior before this update would resize images to an arbitrary ratio. We had removed that resizing but it seems that did not play well with certain cameras and devices. Now the latest code change will resize the image based on the screen width of the device being used. This in turn reduces the size of the image to avoid the crash while also maintaining the aspect ratio of the image.

**Comment by dshokouhi** [comment]
> does this resizing have an impact on image quality ?

We are only scaling down the image so there should not be an impact on image quality.



> And how much time approximately would it take for this fix to be available in Play Store ? (just to have an idea if we're talking hours, days, weeks, ..)

Depends on when the fix is merged. Production release we are not sure when it will happen as we have begun the process of adding the Automotive module which will impact our review process with Google. With that said. Once the PR is merged you have the option to either sideload the latest APK from the releases tab or wait for the next beta version to go live. The fastest turn around would be to install the APK from the releases tab once the fix is merged. The beta version may take some time to get approved for full rollout due to the Automotive work being done.



> This have a sense. My cameras are **5mpx** reolink. I use the substream in widgets, but anyway larger then the device.

Thats the odd part for me. I am using 4k cameras that go up to 8mp and my Pixel 7 Pro has no issues loading it 😄

**Comment by Toukite** [comment]
A month later, in the mood for joking ...

Me waiting for the fix to be released: 
![sad-pablo-lonely](https://user-images.githubusercontent.com/42780622/235071307-d8745e17-e757-4379-bf5c-83fd2d7cfb9d.gif)


And some reading advice for android teams 😅:
![41McQmntVCL _SX397_BO1,204,203,200_](https://user-images.githubusercontent.com/42780622/235071436-ffc6d4a0-b595-44ae-a7a3-2bbe02fee97c.jpg)

**Comment by Toukite** [comment]
Hello,

Following 2023.6 update, I can confirm that the transparency issue is resolved !

I would have however a question related to padding/margin that seems to have changed.
If you take a look at the 3 below screenshots, you'll see a loot of unused space:
![image](https://github.com/home-assistant/android/assets/42780622/8207a68e-8a25-4c57-9910-92d9e5510875)
![image](https://github.com/home-assistant/android/assets/42780622/65276b2d-d945-4244-96b3-5625d3458663)
![image](https://github.com/home-assistant/android/assets/42780622/0ebae6ff-85ce-43d7-bef0-f08de55faa52)

Before the March update and the transparency issue, I was able to have the same 2 full width camera widgets + an additional line of icons, now it's no more possible (the 2 widgets take the full screen).

So my question is: is it something that could be related to this evol, or is it only related to my device ?

Note that I have a grid of 6x5 (Height x Width) and both widgets must now use 3x5 to be full width whereas before I had one that could use only 2x5 to be full width. Device is OnePlus 8 Pro

Thanks in advance for your inputs !

**Comment by Extreme79** [comment]
With the 2024.3.1 the issue seems back to our phones!

**Comment by Palleri** [comment]
Im having this issue aswell.
When I try to use widget for a camera I get a transparent background.

**Comment by dshokouhi** [comment]
> Im having this issue aswell.
> When I try to use widget for a camera I get a transparent background.

This issue is very old please create a new issue and make sure to fill out the issue template completely.
